studio/src/rte/ost.js

```diff
@@ -0,0 +1,234 @@
+    environment: 'PROD',
+    landscape: 'PUBLISHED',
+    language: 'en',
+    searchParameters: {},
+    searchOfferSelectorId: null,
+    defaultPlaceholderOptions: {
+        displayRecurrence: true,
+        displayPerUnit: false,
+        displayTax: false,
+        displayOldPrice: false,
+        forceTaxExclusive: true,
+    },
+    wcsApiKey: 'wcms-commerce-ims-ro-user-cc',
+    ctaTextOption: {
+        ctaTexts: Object.entries(ctaTexts).map(([id, name]) => ({ id, name })),
+        getDefaultText() {
+            return this.ctaTexts[0].id;
+        },
+
+        getTexts() {
+            return this.ctaTexts;
+        },
+
+        getSelectedText(searchParameters) {
+            const ctaLabel = searchParameters.get('text');
+            const selectedText = this.ctaTexts.find(({ id, name }) =>
+                [id, name].includes(ctaLabel),
+            );
+            if (selectedText) return selectedText.id;
+            return ctaLabel || this.getDefaultText();
+        },
+    },
+};
+
+// Function to get the difference between two objects
+function getObjectDifference(values, defaults) {
+    const difference = {};
+
+    // Add properties from values that are different from defaults
+    for (const key in values) {
+        // If the key doesn't exist in defaults, or if the value is different
+        if (!(key in defaults) || values[key] !== defaults[key]) {
+            difference[key] = values[key];
+        }
+    }
+
+    return difference;
+}
+
+export const attributeFilter = (key) =>
+    /^(class|data-|is|href|title)/.test(key);
+
+const OST_TYPE_MAPPING = {
+    price: null,
+    priceStrikethrough: 'strikethrough',
+    priceAnnual: 'annual',
+    priceOptical: 'optical',
+    checkoutUrl: null,
+};
+
+const OST_IS_MAPPING = {
+    price: 'inline-price',
+    priceStrikethrough: 'inline-price',
+    priceAnnual: 'inline-price',
+    priceOptical: 'inline-price',
+    checkoutUrl: 'checkout-link',
+};
+
+const OST_OPTION_ATTRIBUTE_MAPPING = {
+    displayOldPrice: 'data-display-old-price',
+    displayPerUnit: 'data-display-per-unit',
+    displayRecurrence: 'data-display-recurrence',
+    displayTax: 'data-display-tax',
+    forceTaxExclusive: 'data-tax-exclusive',
+    isPerpetual: 'data-perpetual',
+    wcsOsi: 'data-wcs-osi',
+    workflow: 'data-checkout-workflow',
+    workflowStep: 'data-checkout-workflow-step',
+};
+
+export const OST_OPTION_ATTRIBUTE_MAPPING_REVERSE = Object.fromEntries(
+    Object.entries(OST_OPTION_ATTRIBUTE_MAPPING).map(([key, value]) => [
+        value,
+        key,
+    ]),
+);
+
+const OST_OPTION_DEFAULTS = {
+    displayOldPrice: false,
+    displayPerUnit: false,
+    displayRecurrence: true,
+    displayTax: false,
+    forceTaxExclusive: false,
+    isPerpetual: false,
+    workflow: 'UCv3',
+    workflowStep: 'email',
+};
+
+const OST_VALUE_MAPPING = {
+    true: true,
+    false: false,
+};
+
+export function onSelect(offerSelectorId, type, offer, options, promoOverride) {
```

Comment:
`offer` doesn't seem to be used, can we remove it?

Reply:
@Axelcureno we cannot remove, as it is the signature of the callback expected on the OST side.